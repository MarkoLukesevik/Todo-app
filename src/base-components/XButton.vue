<script setup lang="ts">
const emit = defineEmits<{
    (e: 'button-click'): void;
}>()
</script>

<template>
    <button class="x-button" @click="emit('button-click')">
        <img src="/assets/images/icon-cross.svg" alt="">
    </button>
</template>

<style scoped lang="scss">
    .x-button {
        background-color: transparent;
        border: none;
        cursor: pointer;
    }
</style>