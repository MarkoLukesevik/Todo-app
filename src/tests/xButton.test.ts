import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import XButton from '@/base-components/XButton.vue';

describe('Test suite for x button component', () => {
    let wrapper: ReturnType<typeof shallowMount>;

    beforeEach(() => {
        wrapper = shallowMount(XButton);
    });

    it('Should match the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should render correctly', () => {
        expect(wrapper.find('button').exists()).toBe(true);
        expect(wrapper.find('img').attributes('src')).toBe('/assets/images/icon-cross.svg');
    });

    it('Should emit button-click event when button is clicked', async () => {
        await wrapper.find('button').trigger('click');
        expect(wrapper.emitted('button-click')).toBeTruthy();
        expect(wrapper.emitted('button-click')?.length).toBe(1);
    });
});