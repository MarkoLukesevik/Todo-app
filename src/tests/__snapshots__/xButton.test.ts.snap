// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test suite for x button component > Should match the snapshot 1`] = `
<button
  class="x-button"
  data-v-896068b5=""
>
  <img
    alt=""
    data-v-896068b5=""
    src="/assets/images/icon-cross.svg"
  />
</button>
`;
